registry/storage/blobwriter_resumable.go

```diff
@@ -52,20 +55,26 @@ func (bw *blobWriter) resumeDigest(ctx context.Context) error {
 
 	if hashStateMatch.offset == 0 {
 		// No need to load any state, just reset the hasher.
-		h.Reset()
+		h.(hash.Hash).Reset()
 	} else {
 		storedState, err := bw.driver.GetContent(ctx, hashStateMatch.path)
 		if err != nil {
 			return err
 		}
 
-		if err = h.Restore(storedState); err != nil {
+		// This type assertion is safe since we already did an assertion at the beginning
+		if err = h.(encoding.BinaryUnmarshaler).UnmarshalBinary(storedState); err != nil {
```

Comment:
Is the type assertion necessary? `h` is already asserted to be a `encoded.BinaryMarsheler`, right?

Reply:
This is a `encoding.BinaryUnmarshaler`

Reply:
Oh derp 😛